specification/mcpp/resource-manager/Microsoft.CloudPartnerProgramMembership/preview/2023-02-09-preview/mcpp.json

```diff
@@ -0,0 +1,447 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-02-09-preview",
+    "title": "McppClient",
+    "description": "MCPP is the newest program set to track partners, measure their success, and qualify and reward them as Solution Partners across seven different designation groups.  The McppClient provides an interface to create an manage MCPP related purchases programatically"
+  },
+  "host": "management.azure.com",
+  "schemes": ["https"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "security": [
+    {
+      "azure_auth": ["user_impersonation"]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.CloudPartnerProgramMembership/operations": {
+      "get": {
+          "tags": [
+              "Operations"
+          ],
+          "description": "Lists all of the available RP operations.",
+          "operationId": "Operations_List",
+          "x-ms-examples": {
+            "getOperations": {
+              "$ref": "./examples/getOperations.json"
+            }
+          },
+          "parameters": [{
+              "$ref": "#/parameters/ApiVersionParameter"
+          }],
+          "responses": {
+              "200": {
+                  "description": "OK. The request has succeeded.",
+                  "schema": {
+                      "$ref": "#/definitions/OperationListResult"
+                  }
+              },
+              "default": {
+                  "description": "Resource Provider error response describing why the operation failed.",
+                  "schema": {
+                      "$ref": "#/definitions/ErrorResponse"
+                  }
+              }
+          },
+          "x-ms-pageable": {
+              "nextLinkName": "nextLink"
+          }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CloudPartnerProgramMembership/cloudPartnerProgramMemberships": {
```

Comment:
I'd avoid repeating "cloudPartnerProgramMembership" in resource type name... How about:

namespace = Microsoft.CloudPartnerProgram
resource type = membership (note: membership in plural form can still be membership)

So: .../providers/Microsoft.CloudPartnerProgram/membership

Reply:
yes please dont repeat the name between the namespace and resource type

Reply:
ok, will make this change.